erc1155/src/lib.rs

```diff
@@ -205,36 +208,51 @@ impl ExtendERC1155TokenBase for ERC1155Token {
         if ids.len() != amounts.len() {
             panic!("ERC1155: ids and amounts length mismatch")
         }
+        ids.iter()
+            .enumerate()
+            .for_each(|(_s, x)| self.mint(account, x, amounts[_s]));
+    }
 
-        for (i, ele) in ids.iter().enumerate() {
-            let amount = amounts[i];
-            let prev_balance = self.balance_of(account, ele);
-            self.set_balance(account, ele, prev_balance.saturating_add(amount));
+    fn burn(&mut self, id: &u128, amount: u128) {
+        let owner = &msg::source();
+        if self.can_burn(owner, id, amount) {
+            self.set_balance(
+                &msg::source(),
+                id,
+                self.get_balance(owner, id).saturating_sub(amount),
+            );
         }
     }
 
     fn burn_batch(&mut self, ids: &[u128], amounts: &[u128]) {
-        let owner = &msg::source();
-
         if ids.len() != amounts.len() {
             panic!("ERC1155: ids and amounts length mismatch")
         }
 
-        if !self.owner_of_batch(ids) {
-            panic!("ERC1155: have no ownership of ids")
-        }
-
-        for (i, ele) in ids.iter().enumerate() {
-            let amount = amounts[i];
+        ids.iter().enumerate().for_each(|(_s, x)| {
+            if !self.can_burn(&msg::source(), x, amounts[_s]) {
+                panic!("ERC1155: all batch element should be burnable");
+            }
+        });
 
-            let owner_balance = self.balance_of(owner, ele);
+        ids.iter()
+            .enumerate()
+            .for_each(|(_s, x)| self.burn(x, amounts[_s]));
+    }
 
-            if owner_balance < amount {
-                panic!("ERC1155: burn amount exceeds balance")
-            }
+    fn uri(&self, id: u128) -> String {
+        self.base_uri.clone().replace("{id}", &format!("{}", id))
+    }
 
-            self.set_balance(owner, ele, owner_balance.saturating_sub(amount));
+    fn can_burn(&mut self, owner: &ActorId, id: &u128, amount: u128) -> bool {
+        if !self.owner_of(id) {
+            return false;
         }
+        let owner_balance = self.balance_of(owner, id);
+        if owner_balance < amount {
+            return false;
+        }
+        true
```

Comment:
Maybe it is better to make it in a more compact way?

```suggestion
        if !self.owner_of(id) || self.balance_of(owner, id) < amount {
            return false;
        }
        
        true
```

Reply:
Or even 😄 

```suggestion
        self.owner_of(id) && self.balance_of(owner, id) >= amount
```

Reply:
Fixed.